File: week6-after-admission/Find Target Indices After Sorting Array.py
class Solution:
    def targetIndices(self, nums: List[int], target: int) -> List[int]:
        nums.sort()
        
        indexes = []
        for i in range(len(nums)):
            num = nums[i]
            if num == target:
                indexes.append(i)
        return indexes
    
#10min
#1sub

File: week1/Find Target Indices After Sorting Array.js
var targetIndices = function (nums, target) {
	return getTargetIndicesAfterSorting(nums, target);
};

function getTargetIndicesAfterSorting(numbers, target) {
	const sortedNumbers = [...numbers];
	sortedNumbers.sort((a, b) => a - b);

	const matcher = (element) => element === target;
	const firstTargetIndexInSorted = sortedNumbers.findIndex(matcher);

	const targetIndices = [];
	for (let i = firstTargetIndexInSorted; i < sortedNumbers.length; i++) {
		const elementAtIndex = sortedNumbers[i];
		if (elementAtIndex === target) targetIndices.push(i);
		else break;
	}

	return targetIndices;
}
